galaxy_importer/models.py

```diff
@@ -0,0 +1,393 @@
+
+    authors = attr.ib(factory=list)
+    tags = attr.ib(factory=list)
+
+    license_file = attr.ib(
+        default=None,
+        validator=attr.validators.optional(attr.validators.instance_of(str)))
+    readme = attr.ib(
+        default=None,
+        validator=attr.validators.optional(attr.validators.instance_of(str)))
+
+    dependencies = attr.ib(
+        factory=dict,
+        converter=convert_none_to_empty_dict,
+        validator=attr.validators.instance_of(dict))
+
+    @property
+    def label(self):
+        return f"{self.namespace}.{self.name}"
+
+    @staticmethod
+    def value_error(msg):
+        raise ValueError(f"Invalid collection metadata. {msg}") from None
+
+    @namespace.validator
+    @name.validator
+    @version.validator
+    def _check_required(self, attribute, value):
+        """Check that value is present."""
+        if not value:
+            self.value_error(f"'{attribute.name}' is required")
+
+    @namespace.validator
+    @name.validator
+    def _check_name(self, attribute, value):
+        """Check value against name regular expression."""
+        if not re.match(constants.NAME_REGEXP, value):
+            self.value_error(f"'{attribute.name}' has invalid format: {value}")
+
+    @version.validator
+    def _check_version_format(self, attribute, value):
+        """Check that version is in semantic version format."""
+        if not semantic_version.validate(value):
+            self.value_error(
+                "Expecting 'version' to be in semantic version "
+                f"format, instead found '{value}'.")
+
+    @authors.validator
+    @tags.validator
+    @license.validator
+    def _check_list_of_str(self, attribute, value):
+        """Check that value is a list of strings."""
+        err_msg = "Expecting '{attr}' to be a list of strings"
+        if not isinstance(value, list):
+            self.value_error(err_msg.format(attr=attribute.name))
+        for list_item in value:
+            if not isinstance(list_item, str):
+                self.value_error(err_msg.format(attr=attribute.name))
+
+    @license.validator
+    def _check_licenses(self, attribute, value):
+        """Check that all licenses in license list are valid."""
+        # load or return already loaded data
+        valid_license_ids = spdx_licenses.get_spdx()
+
+        invalid_licenses = [license_id for license_id in value
+                            if not self._is_valid_license_id(
+                                license_id, valid_license_ids)]
+        if invalid_licenses:
+            self.value_error(
+                "Expecting 'license' to be a list of valid SPDX license "
+                "identifiers, instead found invalid license identifiers: '{}' "
+                "in 'license' value {}. "
+                "For more info, visit https://spdx.org"
+                .format(', '.join(invalid_licenses), value))
+
+    @staticmethod
+    def _is_valid_license_id(license_id, valid_license_ids):
```

Comment:
Looks like this function is generic. It can be moved to `spdx_licenses.py`.

Reply:
Updated